### ISSUE TITLE
Adding support for basic authentification

### ISSUE THREAD

**Issue by p3l1** [pan-net-security/certbot-dns-acmedns#2]
I am planning to use the acme-dns service behind an nginx reverse proxy with basic authentification, so the api is not exposed on the internet directly.

Because there will be many clients using the api, switching the `/register` endpoint on and off is not a sufficient solution for my usecase.

Would you accept a PR for this feature?

**Comment by mbag** [comment]
Hi @p3l1 , yes I/we will accept the PR for this feature.

**Comment by ZPascal** [comment]
@p3l1 Good idea. After the implementation of the new feature, I can remove my custom validation [script](https://gist.github.com/ZPascal/597da0ac4fb3adec159071435790cff9), the `--manual-auth-hook` call inside `certbot certonly --manual-public-ip-logging-ok --manual --manual-auth-hook /srv/test/acme-dns-client.py --preferred-challenges=dns --email webmaster@test.com --server https://acme-v02.api.letsencrypt.org/directory --agree-tos -d ...... ` command and directly using the plugin.
